src/mpl/src/thread/mpl_thread.c

```diff
@@ -6,7 +6,13 @@
 
 #include "mpl.h"
 
-MPL_SUPPRESS_OSX_HAS_NO_SYMBOLS_WARNING;
+int MPL_is_threaded = -1;
+
+void MPL_set_threaded(int isThreaded)
+{
+    assert(MPL_is_threaded == -1);
+    MPL_is_threaded = isThreaded;
+}
```

Comment:
I haven't run it, but does this logic work if I repeat `{MPI_Init_thread(); MPI_Finalize();}`?

Reply:
MPI standards does not allow such. :)

Reply:
Wow. I didn't know that. Thanks! Then there's no problem.